builtin/logical/pki/dnstest/server.go

```diff
@@ -286,6 +297,9 @@ func (ts *TestServer) AddRecord(domain string, record string, value string) {
 }
 
 func (ts *TestServer) RemoveAllRecords() {
+	ts.lock.Lock()
```

Comment:
Even with the locks in place this RemoveAllRecords won't be safe to call technically in a parallel test context as it's really across the two calls AddXXX and PushConfig or RemoveAll/Push that we should be protecting against. Even that though could cause race issues as the RemoveAll might come in right before another test wants to validate an existing AddRecord.

We could avoid needing the RemoveAll if we had unique identifiers across tests, then the locking as is would probably suffice.

Reply:
@stevendpclark Added, LMKWYT.